File: backend/src/main/java/com/ebikko/mandate/model/ValidationError.java
package com.ebikko.mandate.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ValidationError {

    private final String field;
    private final String value;
    private final String message;

    @JsonCreator
    public ValidationError(@JsonProperty("field") String field,
                           @JsonProperty("value") Object value,
                           @JsonProperty("message") String message) {
        this.field = field;
        this.value = value == null ?  "" : value.toString();
        this.message = message;
    }

    public String getField() {
        return field;
    }

    public String getValue() {
        return value;
    }

    public String getMessage() {
        return message;
    }

}

File: frontend/src/app/model/validationError.ts
import { Serializable } from './serializable';

export class ValidationError implements Serializable<ValidationError> {
	public field: string;
	public value: string;
	public message: string;

	deserialize(input) {
		this.field = input.field;
		this.value = input.value;
		this.message = input.message;
		return this;
	}
}